metron-interface/metron-alerts/src/app/service/rest-api-impl.ts

```diff
@@ -0,0 +1,57 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {Observable} from 'rxjs/Rx';
+import {Headers, RequestOptions} from '@angular/http';
+
+import {HttpUtil} from '../utils/httpUtil';
+import {AlertsSearchResponse} from '../model/alerts-search-response';
+import {SearchRequest} from '../model/search-request';
+import {ElasticSearchLocalstorageImpl} from './elasticsearch-localstorage-impl';
+import {INDEXES} from '../utils/constants';
+import {ColumnMetadata} from '../model/column-metadata';
+import {MetronRestApiUtils} from '../utils/metron-rest-api-utils';
+import {AlertSource} from '../model/alert-source';
+
+export class RestApiImpl extends ElasticSearchLocalstorageImpl {
```

Comment:
Could we just call this SearchService or something similar?  There are going to be other functions we add to the REST layer so this class is going to grow to be very large eventually if we just have one generic REST service.  To me, it would make more sense to create a matching service for each REST controller.  This is how it was done with the management UI.